lib/utils.py

```diff
@@ -29,5 +31,20 @@ def get_pascal_case(in_string):
     return ''.join([s.capitalize() for s in in_string.split('_')])
 
 def get_metric_providers(config):
-    metric_providers_keys = config['measurement']['metric-providers'].keys()
+    architecture = get_architecture()
+    if 'common' in config['measurement']['metric-providers']:
```

Comment:
This function will fail if the "common" key does not contain any active providers.

Reply:
good catch, pushed a check against this (and the same was true if there were no providers under the architecture key as well)